### PR TITLE
Hooking up search results page to GraphQl

### PR DESCRIPTION
**Pull request by rjkerrison**
# Search result page uses GraphQL

Finishes #621. Gives @craiga a nice going away present.

Craig

### Acceptance Criteria

- [ ] feature flag present as query parameter
- [ ] with feature flag `useGraphQl=true`, search results are fetched via the API
- [ ] pagination works properly

### Testing information

`yarn dev`, live your best life

### Pre-flight Checklist

- [ ] Testing aligns with [testing strategy][testing strategy]
- [ ] DUXD review approval
- [ ] Accessibility Reviewed
- [ ] Product owner demo

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/web/src/services/documents-loader.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ const convertDocumentResponseToDocument = ({
 }): IDocument => {
   return {
     activeSubstances: doc.activeSubstances,
-    context: doc.highlights,
+    context: doc.highlights?.join(' … ') || '',
```
Comment (majikandy) [inline medicines/web/src/services/documents-loader.ts:76]: interested to know what this line does...

Comment (rjkerrison) [reply medicines/web/src/services/documents-loader.ts:76]: `highlights` come back from Azure, the parts of the blobs where the search terms appear, and we stitch them together with a little `&hellip;` – I'd missed this previously

**Comment by majikandy** [comment]
Looks good, ran it locally - Off by one on the useGraphQl results... ie with 32 results, page 4 only shows 1 item when it should show 2 and all the results shifted by one (ie first result never shows)

**Comment by rjkerrison** [comment]
Thanks @majikandy – we've corrected the cursor-making logic on the website so that it agrees with the API. That'll give us a working version from this PR, and we can make the change to accept _offset_ as discussed in a future change.

**Comment by majikandy** [comment]
managed to crash the page by putting a high number for the page value

Screenshot 2020-05-01 10 29 18
m/s/yw5ilw0ryedh1gn/Screenshot%202020-05-01%2010.29.18.png?dl=0

File: medicines/web/src/services/search-results-loader.test.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { makeCursor } from './search-results-loader';
+
+describe(makeCursor, () => {
+  it.each([2, 4, 5, 10, 15, 20])('offsets first result by -1', pageSize => {
```
Comment (majikandy) [inline medicines/web/src/services/search-results-loader.test.ts:4]: 👍

File: medicines/web/src/services/search-results-loader.test.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    expect(cursor).toBe('LTE=');
+  });
+
+  it.each([
+    [2, 10, 'OQ=='],
+    [3, 5, 'OQ=='],
+    [2, 2, 'MQ=='],
+    [3, 1, 'MQ=='],
+    [2, 3, 'Mg=='],
+    [4, 1, 'Mg=='],
```
Comment (majikandy) [inline medicines/web/src/services/search-results-loader.test.ts:15]: took my brain a little while to figure out but yes the ones that are the same are due to being a different page size - ie page 3 of size 5 is the 11th item onwards, as is page 2 of size 10

**Pull request merged**